flink/flink-1.13/src/main/java/io/tidb/bigdata/flink/connector/source/enumerator/TiDBSourceSplitEnumerator.java

```diff
@@ -108,23 +111,27 @@ private void assignPendingSplits(Set<Integer> pendingReaders) {
     }
   }
 
+  public Set<TiDBSourceSplit> getSplits() {
+    try (ClientSession splitSession = ClientSession
+        .createWithSingleConnection(new ClientConfig(properties))) {
+      // check exist
+      final String databaseName = properties.get("tidb.database.name");
+      final String tableName = properties.get("tidb.table.name");
+      splitSession.getTableMust(databaseName, tableName);
+      timestamp = splitSession.getTimestamp();
+      final TableHandleInternal tableHandleInternal = new TableHandleInternal(
+          UUID.randomUUID().toString(), databaseName, tableName);
+      List<SplitInternal> splits =
+          new SplitManagerInternal(splitSession).getSplits(tableHandleInternal, timestamp);
+      return splits.stream().map(TiDBSourceSplit::new).collect(Collectors.toSet());
+    } catch (Exception e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
   @Override
   public void start() {
-    context.callAsync(() -> {
-      try (ClientSession splitSession = ClientSession
-          .createWithSingleConnection(new ClientConfig(properties))) {
-        // check exist
-        final String databaseName = properties.get("tidb.database.name");
-        final String tableName = properties.get("tidb.table.name");
-        splitSession.getTableMust(databaseName, tableName);
-        timestamp = splitSession.getTimestamp();
-        final TableHandleInternal tableHandleInternal = new TableHandleInternal(
-            UUID.randomUUID().toString(), databaseName, tableName);
-        List<SplitInternal> splits =
-            new SplitManagerInternal(splitSession).getSplits(tableHandleInternal, timestamp);
-        return splits.stream().map(TiDBSourceSplit::new).collect(Collectors.toSet());
-      }
-    }, this::assignSplits);
+    context.callAsync(() -> allSplits, this::assignSplits);
```

Comment:
Since we have done the most time consuming part, we can move assignSplits into constructor and make start a noop implementation.